internal/shim-kvm/.cargo/config

```diff
@@ -1,10 +1,14 @@
 [build]
-target = "x86_64-unknown-linux-musl"
+# FIXME: "x86_64-unknown-none" is not yet a tier 2 target, so setting this as
+#        the default target breaks rust-analyzer.
+# target = "x86_64-unknown-none"
 
-[target.x86_64-unknown-linux-musl]
+[target.x86_64-unknown-none]
 rustflags = [
     "-C", "relocation-model=pic",
+    "-C", "linker=gcc",
     "-C", "link-args=-Wl,--sort-section=alignment,-Tlayout.ld -nostartfiles",
     "-C", "link-self-contained=no",
     "-C", "force-frame-pointers=yes",
+    "-C", "target-feature=-soft-float,+sse,+sse2",
```

Comment:
@haraldh  is this needed?

Reply:
Not necessarily, the alternative is enabling the `force-soft` feature of `aes-gcm`.

Reply:
somehow I get 
```
LLVM ERROR: Do not know how to split the result of this operator!
```
if I don't specify `-C target-feature=-soft-float,+sse,+sse2,+avx`

For `-C target-feature=+sse,+sse2`, I get:
```
  process didn't exit successfully: `rustc --crate-name compiler_builtins /home/harald/.cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.70/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C panic=abort -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="compiler-builtins"' --cfg 'feature="core"' --cfg 'feature="default"' --cfg 'feature="rustc-dep-of-std"' -C metadata=b0d2b6955f5e2846 -C extra-filename=-b0d2b6955f5e2846 --out-dir /home/harald/git/enarx/enarx/internal/shim-kvm/target/x86_64-unknown-none/debug/deps --target x86_64-unknown-none -Z force-unstable-if-unmarked -L dependency=/home/harald/git/enarx/enarx/internal/shim-kvm/target/x86_64-unknown-none/debug/deps -L dependency=/home/harald/git/enarx/enarx/internal/shim-kvm/target/debug/deps --extern core=/home/harald/git/enarx/enarx/internal/shim-kvm/target/x86_64-unknown-none/debug/deps/librustc_std_workspace_core-82f40907211f5791.rmeta --cap-lints allow -C link-arg=-Tlayout.ld -C link-arg=--sort-section=alignment -C target-feature=+sse,+sse2 --cfg 'feature="unstable"' --cfg 'feature="mem"' --cfg 'feature="mem-unaligned"'`
(signal: 5, SIGTRAP: trace/breakpoint trap)
```

Reply:
@haraldh @Freax13 My point here is that these options **HAVE** to go away in order to merge the crates into a workspace. We can't have any `.cargo/config` in that scenario. So all options **MUST** move to either a `build.rs` or to the Rust target directly.